packages/sdk-core/src/BacktraceCoreClient.ts

```diff
@@ -418,4 +420,51 @@ export abstract class BacktraceCoreClient<
     private static destroy() {
         this._instance = undefined;
     }
+
+    private validateAttributes() {
+        function isNotEmptyString(v: unknown) {
+            if (typeof v === 'string' && v) {
+                return;
+            }
+
+            return 'must be defined and not an empty string';
+        }
+
+        const validators = {
+            application: isNotEmptyString,
+            'application.version': isNotEmptyString,
+        } as const;
+
+        function validate(attributes: Record<string, unknown>) {
+            const errors: Record<string, string> = {};
+            for (const [attribute, validate] of Object.entries(validators)) {
+                const value = attributes[attribute];
+                const error = validate(value);
+                if (error) {
+                    errors[attribute] = error;
+                }
+            }
+            return errors;
+        }
+
+        // Validate scoped attributes first. If they return no errors, there is no point
+        // in checking all attributes, which resolving of may be slower.
+        const scoped = this.attributeManager.get('scoped');
+        const scopedErrors = validate(scoped.attributes);
+        if (!Object.keys(scopedErrors).length) {
+            return;
+        }
+
+        const allAttributes = this.attributeManager.get();
```

Comment:
```suggestion
        const dynamicAttributes = this.attributeManager.get('dynamic');
        const allAttributes = {...dynamicAttributes, ...scoped};
```

Reply:
It may not be so simple. `get` iterates over all attribute providers, and then tests them if they are scoped/dynamic. For instance, if we would have these providers:
```
const provider1 = { type: 'scoped': get: () => ({ a: 'foo' });
const provider2 = { type: 'dynamic': get: () => ({ a: '' });
const provider3 = { type: 'scoped': get: () => ({ a: 'bar' });
```

`allAttributes` would be `{a: ''}`, while `get()` would return `{ a: 'bar' }`.

Reply:
good point!